src/globus_sdk/experimental/globus_app/_validating_token_storage.py

```diff
@@ -108,31 +114,31 @@ def store_token_data_by_resource_server(
     def get_token_data_by_resource_server(self) -> dict[str, TokenData]:
         """
         :returns: A dict of TokenData objects indexed by their resource server
-        :raises: :exc:`TokenValidationError` if any of the token data have expired or
-            do not meet the attached scope requirements.
+        :raises: :exc:`TokenValidationError` to signal a problem with stored token data.
         """
         token_data_by_resource_server = (
             self._token_storage.get_token_data_by_resource_server()
         )
 
         for resource_server, token_data in token_data_by_resource_server.items():
-            self._validate_token_meets_scope_requirements(resource_server, token_data)
+            self._validate_token_data_meets_scope_requirements(
+                resource_server, token_data
+            )
 
         return token_data_by_resource_server
 
-    def get_token_data(self, resource_server: str) -> TokenData | None:
+    def get_token_data(self, resource_server: str) -> TokenData:
         """
         :param resource_server: A resource server with cached token data.
-        :returns: The token data for the given resource server, or None if no token data
-            is present in the attached storage adapter.
-        :raises: :exc:`TokenValidationError` if the token has expired or does not meet
-            the attached scope requirements.
+        :returns: The token data for the given resource server.
+        :raises: :exc:`TokenValidationError` to signal a problem with stored token data.
```

Comment:
```suggestion
        :raises: :exc:`MissingTokensError` to signal a problem with stored token data.
```

Reply:
`TokenValidationError` is the base subclass of errors raised by this method (of which `MissingTokensError` is one) and multiple different `TokenValidationError` types can be raised from this method.

The basic error heirarchy structure is
  * "TokenValidationError" - category of errors
     * `MissingTokensError` - No token exists for resource server.
     * `ExpiredTokenError` - Token exists but is expired.
     * `UnmetScopeRequirementsError` - Token exists but root or dependent scope requirements are unmet.
     * "IdentityMismatchError" - category of errors
        * `MissingIdentityError` - Token response does not include identity info.
        * `IdentityMismatchError` - Token response includes identity info but it differs from the stored identity.

With the latest rewrite, `get_token_error` does only raise 2 of these error types: [`MissingTokensError`, `UnmetScopeRequirementsError`] (`ExpiredTokenError` was pulled based on a discussion Aaron & I had about how RefreshAuthorizerFactory should function in the case of an expired access token) so I suppose I could scope the docstring down to just those two?

Reply:
Updated this & other docstrings to define more concretely which errors they raise.